components/apps_form/apps_form_field/apps_form_field.tsx

```diff
@@ -45,10 +45,16 @@ export default class AppsFormField extends React.PureComponent<Props> {
     };
 
     handleSelected = (selected: AppSelectOption | UserProfile | Channel) => {
```

Comment:
I think the method signature should be

`handleSelected = (selected: AppSelectOption | AppSelectedOption[])`

This should allow the type system to help with array vs object types

Reply:
@javaguirre We shouldn't expect to receive `UserProfile` or `Channel` here though. For the multi-select  channel fields we would technically be receiving `(AppSelectOption & Channel) | (AppSelectOption & Channel)[]`, though this method only requires the fields from `AppSelectOption`, so `AppSelectOption | AppSelectOption[]` should cover all cases.